### DISCUSSION TITLE
Issues with chatgpt template

### DISCUSSION THREAD

**Discussion by sumukhshetty** [AntlerVC/firetable#1260] [Support / Q&A] [answered]
I've been trying to get the chatbot rowy whatsapp template running but I keep hitting bottlenecks with it. 
The Twilio integration seems to be working as my messages are getting logged, but there are no replies being generated - my openai key seems to be fine - but it doesnt seem like rowy has even connected with Openai as the secret status shows as never used 

Can someone help ? Not sure what I'm doing wrong. I get the Ai is thinking message and in the extension log I see something to do with reply but there is no openai response generation happening.

**Comment by sumukhshetty** [comment]
Attaching a image of my whatsapp 


<img width="534" alt="Screenshot 2023-05-13 at 4 02 10 PM" src="https://github.com/rowyio/rowy/assets/22879373/99d16a8d-626c-48c1-be11-4d166463c99a">

**Reply by harinij** [reply, marked as answer]
Hi @sumukhshetty could you check a few things to troubleshoot this further:
1) Right click on the derivative and see if the `eval` works? 
2) Another thing to make sure is if your OpenAI key is added correctly in Google Cloud Secret manager. 
3) In addition, in the derivative column config code, you can also add `logging.log` statements to see if the function is getting triggered. These log statements will show up in the [cloud logs](https://docs.rowy.io/cloud-logs) section.
